test/parallel_api/algorithm/alg.modifying.operations/transform_unary.pass.cpp

```diff
@@ -53,9 +35,51 @@ struct test_one_policy
         EXPECT_TRUE(out_last == orr, "transform returned wrong iterator");
         check_and_reset(first, last, out_first);
     }
+
+private:
+    template <typename InputIterator, typename OutputIterator>
+    void
+    check_and_reset(InputIterator first, InputIterator last, OutputIterator out_first)
+    {
+        typename ::std::iterator_traits<OutputIterator>::difference_type k = 0;
+        for (; first != last; ++first, ++out_first, ++k)
+        {
+            // check
+            const auto expected = get_expected(*first);
+            auto& actual = get_actual(*out_first);
+            EXPECT_EQ(expected, actual, "wrong value in output sequence");
+            // reset
+            actual = k % 7 != 4 ? 7 * k - 5 : 0;
+        }
+    }
+    template <typename InputT>
+    auto
+    get_expected(InputT val)
+    {
+        return 1 - val;
```

Comment:
optional nitpick:

Since we are hardcoding the "expected" calculation here to be a complement operation, should we rename the `UnaryComplementOp` and `_ComplementOp` below to make it clear that arbitrary unary functions aren't appropriate for use here?  These template arguments are merely to support different types of complement operations.

Reply:
I "joined" operation and get_expected/get_actual in one stuff.